Code quality analysis for pull requests

It can be helpful to have a tool checking the code for syntax errors or possible pitfalls.

[Shellcheck](https://www.shellcheck.net/) is a great tool for doing exactly that on shell scripts.

Here's an example:
```sh
% shellcheck terramorf

In terramorf line 32:
VERSION=0.5.2
^-----^ SC2034 (warning): VERSION appears unused. Verify use (or export if used externally).

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- VERSION appears unused. Verify us...
```

A GitHub Actions workflow running this tool against pull requests might interest you.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -1,0 +1,18 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Shellcheck
+        run: shellcheck --shell=bash --severity=warning install.sh terramorf

